src/Service/RegenerateProductUrl.php

```diff
@@ -134,6 +134,19 @@ public function execute(?array $productIds = null, ?int $storeId = null, bool $v
                 if (count($newUrls)) {
                     $regeneratedForStore += $this->replaceUrls($newUrls, true);
                 }
+            } catch (\Magento\UrlRewrite\Model\Exception\UrlAlreadyExistsException $e) {
+                $this->log(sprintf(
+                    '<error>Couldn\'t insert duplicate URL rewrites for the following ' .
+                    'products on store ID %d:' . PHP_EOL . '%s</error>',
+                    $store->getId(),
+                    implode(PHP_EOL, array_map(function ($url) {
+                        return sprintf(
+                            '- Product ID: %d, request path: %s',
+                            $url['entity_id'],
+                            $url['request_path']
+                        );
+                    }, $e->getUrls()))
+                ));
```

Comment:
/wtd Clarify the log message to make it clear that even one error will cause the entire batch of ~500 URLs to be rolled back and fail.